Build with `MEMFENCE` failed

```sh
$ make
    CC julia-w64/src/gc-debug.dbg.obj
D:/jl/julia/src/gc-debug.c: In function 'gc_verify_tags_pagetable0':
D:/jl/julia/src/gc-debug.c:357:35: error: 'pagetable0_t' has no member named 'allocmap'
  357 |         uint32_t line = pagetable0->allocmap[pg_i];
      |                                   ^~
D:/jl/julia/src/gc-debug.c:361:57: warning: passing argument 1 of 'gc_verify_tags_page' makes pointer from integer without a cast [-Wint-conversion]
  361 |                     gc_verify_tags_page(pagetable0->meta[pg_i * 32 + j]);
      |                                         ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
      |                                                         |
      |                                                         uint8_t {aka unsigned char}
D:/jl/julia/src/gc-debug.c:290:51: note: expected 'jl_gc_pagemeta_t *' {aka 'struct _jl_gc_pagemeta_t *'} but argument is of type 'uint8_t' {aka 'unsigned char'}
  290 | static void gc_verify_tags_page(jl_gc_pagemeta_t *pg)
      |                                 ~~~~~~~~~~~~~~~~~~^~
D:/jl/julia/src/gc-debug.c: In function 'gc_verify_tags_pagetable1':
D:/jl/julia/src/gc-debug.c:371:35: error: 'pagetable1_t' has no member named 'allocmap0'
  371 |         uint32_t line = pagetable1->allocmap0[pg_i];
      |                                   ^~
D:/jl/julia/src/gc-debug.c: In function 'gc_verify_tags_pagetable':
D:/jl/julia/src/gc-debug.c:385:25: error: 'memory_map' undeclared (first use in this function)
  385 |         uint32_t line = memory_map.allocmap1[pg_i];
      |                         ^~~~~~~~~~
D:/jl/julia/src/gc-debug.c:385:25: note: each undeclared identifier is reported only once for each function it appears in
make[1]: *** [/d/jl/julia/src/Makefile:241: gc-debug.dbg.obj] Error 1
make: *** [/d/jl/julia/Makefile:97: julia-src-debug] Error 2
```

https://github.com/JuliaLang/julia/blob/eb26d638db8f1651be38492ab8df5fe037cfd80c/src/gc.h#L267-L278
> error: 'pagetable0_t' has no member named 'allocmap'
https://github.com/JuliaLang/julia/blob/eb26d638db8f1651be38492ab8df5fe037cfd80c/src/gc-debug.c#L352-L355
> error: 'pagetable1_t' has no member named 'allocmap0'
https://github.com/JuliaLang/julia/blob/eb26d638db8f1651be38492ab8df5fe037cfd80c/src/gc-debug.c#L366-L369

> error: 'memory_map' undeclared
https://github.com/JuliaLang/julia/blob/ca1e51984b620e291ef27fb2484512f3c2fe0cac/src/gc-debug.c#L380-L392

--- a/src/gc-debug.c
+++ b/src/gc-debug.c
@@ -349,44 +349,15 @@ static void gc_verify_tags_page(jl_gc_pagemeta_t *pg)
     }
 }
 
-static void gc_verify_tags_pagetable0(pagetable0_t *pagetable0)
+static void gc_verify_tags_pagestack(void)
 {
-    for (int pg_i = 0; pg_i < REGION0_PG_COUNT / 32; pg_i++) {
-        uint32_t line = pagetable0->allocmap[pg_i];
-        if (line) {
-            for (int j = 0; j < 32; j++) {
-                if ((line >> j) & 1) {
-                    gc_verify_tags_page(pagetable0->meta[pg_i * 32 + j]);
-                }
-            }
-        }
-    }
-}
-
-static void gc_verify_tags_pagetable1(pagetable1_t *pagetable1)
-{
-    for (int pg_i = 0; pg_i < REGION1_PG_COUNT / 32; pg_i++) {
-        uint32_t line = pagetable1->allocmap0[pg_i];
-        if (line) {
-            for (int j = 0; j < 32; j++) {
-                if ((line >> j) & 1) {
-                    gc_verify_tags_pagetable0(pagetable1->meta0[pg_i * 32 + j]);
-                }
-            }
-        }
-    }
-}
-
-static void gc_verify_tags_pagetable(void)
-{
-    for (int pg_i = 0; pg_i < (REGION2_PG_COUNT + 31) / 32; pg_i++) {
-        uint32_t line = memory_map.allocmap1[pg_i];
-        if (line) {
-            for (int j = 0; j < 32; j++) {
-                if ((line >> j) & 1) {
-                    gc_verify_tags_pagetable1(memory_map.meta1[pg_i * 32 + j]);
-                }
-            }
+    for (int i = 0; i < gc_n_threads; i++) {
+        jl_ptls_t ptls2 = gc_all_tls_states[i];
+        jl_gc_page_stack_t *pgstk = &ptls2->page_metadata_allocd;
+        jl_gc_pagemeta_t *pg = jl_atomic_load_relaxed(&pgstk->bottom);
+        while (pg != NULL) {
+            gc_verify_tags_page(pg);
+            pg = pg->next;
         }
     }
 }
@@ -423,7 +394,7 @@ void gc_verify_tags(void)
 
     // verify that all the objects on every page are either valid julia objects
     // or are part of the freelist or are on the allocated half of a page
-    gc_verify_tags_pagetable();
+    gc_verify_tags_pagestack();
 }
 #endif
 

